### DISCUSSION TITLE
How are we going to handle `Internet Explorer` retirement?

### DISCUSSION THREAD

**Discussion by OnkarRuikar** [mdn/mdn-community#620] [Content] [answered]
Internet Explorer is reaching end of life soon(~2 months): https://death-to-ie11.com

How are we going to handle it on MDN?
1. Remove the `Internet Explorer` column from BCD tables?
2. Remove all API pages specific to the IE? such as https://developer.mozilla.org/en-US/docs/Web/API/MSCandidateWindowHide
3. Remove all contents related to the IE? such as, on page [Design for all types of users](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Design_for_all_types_of_users)
> Yet, for many years Internet Explorer adamantly displayed 16 pixels as 16 pixels. In this case, zooming did nothing, even as late as Internet Explorer 8, which we still have to cater to because it's still around.
4. Gradually remove all mentions of it from `mdn/content`, and leave absolutely no trace of it?
5. An info/warning message, to developers, on home page or a banner on top?

Note: there are around 472 cases of point 2 and 3 in `mdn/content`

**Comment by teoli2003** [comment]
As the end-of-life for IE11 is not really complete in 2 months (only on the most common products), I think we can start by doing this:

1. _Hide by default_ the IE column in BCD tables. Don't touch the info in mdn/browser-compat-data yet.
2. Content related to IE:
 - Keep the API pages specific to IE11. Unlink them from the standard interface (text and sidebars), but keep them linked in: https://developer.mozilla.org/en-US/docs/Web/API/Microsoft_Extensions (that can be renamed)
 - Remove any API page specific to IE10 or earlier.
3. (and 4.) Remove all mentions of IE in the prose of regular articles (reference, guides, and LA). (Clean MDN)

I don't think there is a  need for a site-wide message to developers. 

That way: we clean our content and make it easier (= contained) to remove what is left when we decide to do it while making it harder (= more costly) to maintain for IE11, and impossible (on MDN) for earlier IE.

I think none of this work would be thrown away when we remove all mentions of IE (the worst-case scenario is in 2029).

cc/ @ddbeck @rumyra @wbamberg @Elchi3 @schalkneethling

**Reply by ddbeck** [reply]
I think this is a good overview. One thing I'd add is about timing: I would not regard June 2022 as a point in time when it is safe to start removing IE mentions. Rather, I would expect that to be a point in time when many developers may be in the process of upgrading various applications or finally able to start using features not supported by IE. That is to say, the IE information will continue to be valuable even past the EOL date, as a point of comparison for those developers.

I'd suggest waiting 6 months or so before dropping IE-exclusive content outright or removing IE from compat tables. I'd suggest waiting the full 2 years to remove it from BCD (or until such time as BrowserStack and other tools stop offering the ability to test these things), in line with BCD's other irrelevance guidelines.

**Comment by Seirdy** [comment]
Worth remembering that Edge will still have IE-mode and that I imagine most current IE users who weren't dissuaded by the "switch to Edge!" banners will suddenly switch.

I want to drop IE too, but 5% of respondents to the latest WebAIM screen reader survey use it (I imagine the real percentage is higher since survey respondents are not representative), so I think that the right course of action would be to avoid completely erasing it. That is, at *least* until its market share drops not just globally but in each population MDN wishes to support: Japan and the screen reader community see disproportionately high usage.

Annotating all IE-specific information and including links warning users to switch would be more helpful.

I also don't think the information should "disappear" or become unnecessarily hard to find; it just shouldn't crowd the more relevant information. A `<details>` element with a  "toggle information for discontinued browsers" `<summary>` button could be one of many ways to go about this; the contained text could lead with the fact that the covered browsers are discontinued and unsupported, and information is no longer being updated to meet MDN's standards. The IE column in BCD tables could have a "discontinued" label but still show.

Preserving some historical information to show progress over time is really helpful. I like knowing not only which features are supported, but which versions of each browser were the first to get support. IE information could hold comparable value to some.

Showing IE info doesn't need to imply any sort of endorsement (far from).

Relevant discussion: https://github.com/mdn/browser-compat-data/issues/15333

**Reply by hamishwillee** [reply]
I like this, in particular the point that it is not necessary to have a hard decision around cleaning BCD - we can hide or toggle-display the rendered version. I think we might reasonably make the information less prominent far earlier than we delete it.

**Comment by Rumyra** [comment]
I agree about not doing this immediately but setting a time in the future to start changes - @ddbeck 's 6 months and 2 years sounds good to me 👍

**Comment by estelle** [comment]
IE has been removed from the presented BCD tables.
Several PRs were merged to remove mention of IE that weren't historical in nature, including https://github.com/mdn/content/pull/21718, and https://github.com/mdn/content/pull/21721, along with some that remove legacy fallback support (example: https://github.com/mdn/content/pull/31473). 

Our documentation still includes some references to IE, mostly historical, including when some media codecs initially got browser support. I created two PRs to get rid of most of these: https://github.com/mdn/content/pull/32868/ and https://github.com/mdn/content/pull/32861.

What should be done about IE/Edge to 18 features like Web/API/MediaError/msExtendedCode?

**Reply by estelle** [reply]
Ah, reread the thread!

Next month we can remove all MS features like  https://developer.mozilla.org/en-US/docs/Web/API/MSCandidateWindowHide and  https://developer.mozilla.org/en-US/docs/Web/API/MediaError/msExtendedCode!

**Comment by estelle** [comment]
Let's write an issue to remove all msXX API features starting in April and close this discussion as resolved :D

**Reply by estelle** [reply]
there weren't that many that I found:

https://github.com/mdn/content/issues/32929

**Reply by teoli2003** [reply, marked as answer]
When this mdn/content#33012 is merged, we can mark this as _answered_: the last bits of IE-related documentaiton will no more be on mdn/content.

There are years that mdn/content was not suitable for IE-specific stuff, and these "exceptions" were going on the way of other developers by making the web platform look more complicated that it is nowadays.

**Comment by Seirdy** [comment]
I don't think I see the merit of *deleting* historical knowledge, rather than simply putting a notice that the feature is obsolete. Are there any other easily-accessible resources with as much detail on historical browser features?

**Reply by estelle** [reply]
most have already been deleted. 
All pages that exist must be maintained. GitHub does keep our history though, so we can delete while still leaving a historical trail.

**Reply by Seirdy** [reply]
I see. Is anybody interested in forking the repo before this happens and hosting historical information before it's deleted yet again? Something similar happened the last one or two times MDN purged historical/unpopular information instead of marking them as obsolete; I think one was for XUL.

**Reply by hamishwillee** [reply]
IMO after a while things marked as obsolete are just clutter for the vast majority of readers. 

But anyhow, there is no need to fork the repo - nothing is ever deleted from here. What would make sense though is to add a tag release or branch back when IE was fully documented. Because if you do need this stuff that would make it a lot easier to find.

**Reply by estelle** [reply]
Whenever a page is deleted, the history is updated, so you can see what the original link was to look at archive.org or github history. https://github.com/mdn/content/blob/main/files/en-us/_wikihistory.json and https://github.com/mdn/content/blob/main/files/en-us/_redirects.txt are huge files, but that's where to find stuff, and searchable.
